controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -210,6 +210,18 @@ func (r *BareMetalHostReconciler) Reconcile(request ctrl.Request) (result ctrl.R
 	}
 	prov, err := r.ProvisionerFactory(host, *bmcCreds, info.publishEvent)
 	if err != nil {
+		if !info.host.DeletionTimestamp.IsZero() {
```

Comment:
That would catch much more than just a malformed BMC address, no ? Basically any errors (even temporary) in the provisioner would lead to BMH deletion, is this a desired behaviour ?